src/main/resources/avro/references.avdl

```diff
@@ -27,8 +105,22 @@ record Reference {
   string name;
 
   /**
+  Start and end locations of how this reference fits into the existing graph.
+  May be null only for a primary sequence that is not joined onto any other.
+  Note that the referenceId for start and end may not be the same.
+  */
+  union { null, ReferenceJoinLocation } startJoin, endJoin;
+
+  /**
+  The sequence to insert between startJoin and endJoin. Must be set if sourceURI
+  is null.
+  */
+  union { null, string } sequence = null;
```

Comment:
This is a good plan - `ListReferenceBases` should be able to take a `Path` as an argument, which would give us the required semantics I think. +1 for discussing the issues around `ListReferenceBases` in #230.

I'm still not sure I see the point in `sourceURI`, but this applies equally to the main branch, so this should be brought up elsewhere.